src/components/accessible-autocomplete/accessible-autocomplete.js

```diff
@@ -52,22 +52,42 @@ AccessibleAutoComplete.prototype.init = function init() {
 
     const selectElementAriaDescribedBy = selectElement.getAttribute('aria-describedby') || '';
     const autocompleteElement = document.getElementById(autocompleteId);
-    const autocompleteElementAriaDescribedBy = (autocompleteElement && autocompleteElement.getAttribute('aria-describedby')) || '';
+    const autocompleteElementAriaDescribedBy = (autocompleteElement?.getAttribute('aria-describedby')) || '';
     const autocompleteElementMissingAriaDescribedAttrs = (
       autocompleteElement
       && autocompleteElement.tagName !== 'select'
       && !autocompleteElementAriaDescribedBy.includes(selectElementAriaDescribedBy)
     );
     if (autocompleteElementMissingAriaDescribedAttrs) {
       // if there is a hint and/or error then the autocomplete element
-      // needs to be aria-describedby these, which it isn't be default
-      // we need to check if it hasn't already been done to avoid
+      // needs to be aria-describedby these, which it isn't be default.
+      // we need to check if it hasn't already been done to avoid adding
+      // them twice if someone has added a separate patch.
       autocompleteElement.setAttribute(
         'aria-describedby',
         `${selectElementAriaDescribedBy} ${autocompleteElementAriaDescribedBy}`,
       );
+
+      if (window.MutationObserver != null) {
+        // when the input is empty, the autocomplete adds a link to a hint
+        // that explains how to interact with the input via aria-describedby
+        // and when it's not empty it removes it. These changes cause the
+        // removal of the links to the error and hint, so we need to add
+        // those links back, as well as maintain the link to the hint if it
+        // was present because the input is empty.
+        new MutationObserver(() => {
+          const currentAriaDescribedBy = autocompleteElement.getAttribute('aria-describedby') || '';
+          if (!currentAriaDescribedBy?.includes(selectElementAriaDescribedBy)) {
+            autocompleteElement.setAttribute('aria-describedby', `${selectElementAriaDescribedBy} ${currentAriaDescribedBy}`);
+          }
+        }).observe(autocompleteElement, {
+          attributes: true,
+          attributeFilter: ['aria-describedby'],
+        });
```

Comment:
it's important to have the check to see if it's already ok - so that you don't get an infinite loop of updates triggering more updates